Assets/Scripts/ButtonListener.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using OculusSampleFramework;
+using UnityEngine.Events;
+
+public class ButtonListener : MonoBehaviour
+{
+    public UnityEvent proximityEvent;
+    public UnityEvent contactEvent;
+    public UnityEvent actionEvent;
+    public UnityEvent defaultEvent;
+
+    void Start()
+    {
+        GetComponent<ButtonController>().InteractableStateChanged.AddListener(InitiateEvent);
+    }
+
+    void InitiateEvent(InteractableStateArgs state)
+    {
+        if (state.NewInteractableState == InteractableState.ProximityState)
```

Comment:
this would probably be better as a switch statement

Reply:
I removed this file